justice/features/period_distribution.py

```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf-8 -*-
+"""Runs various Lomb-Scarble methods."""
+import abc
+import collections
+import typing
+
+import numpy as np
+
+from justice import lightcurve
+import astropy.stats
+from gatspy import periodic
+
+
+class MultiBandPeriod(collections.OrderedDict):
+    __slots__ = (
+        "period",
+        "best_period",
+    )
+
+    def __init__(self, *, period, band_to_power, best_period=None):
+        super().__init__(band_to_power)
+        # for value in self.values():
+        #     assert isinstance(value, np.ndarray)
+        #     assert len(value.shape) == 1, "Expected 1D shape per band."
+        self.period = period
+        self.best_period = best_period
+
+    def plot(self, band_name):
+        import matplotlib.pyplot as plt
+        plt.plot(self.period, self[band_name])
+        plt.xlabel("Period")
+        plt.ylabel("Power")
+
+    def period_max(self, band_name):
+        return self.period[np.nanargmax(self[band_name])]
+
+    def frequency_max(self, band_name):
+        return 1.0 / self.period_max(band_name)
+
+
+class LsTransformBase(object):
+    __metaclass__ = abc.ABCMeta
+
+    def __init__(self, period: typing.Union[str, np.ndarray] = 'default') -> None:
+        if period == 'default':
+            self.period = np.linspace(2., 500., 10000)
+        else:
+            assert isinstance(period, np.ndarray)
+            self.period = period
+
+    @abc.abstractmethod
+    def apply(self, lc: lightcurve._LC) -> MultiBandPeriod:
+        raise NotImplementedError()
+
+
+def _compute_ls(band, period):
+    return astropy.stats.LombScargle(band.time, band.flux, band.flux_err).power(
+        period
+    )  # replace with autopower soon
+
+
+class IndependentLs(LsTransformBase):
```

Comment:
Can we mark these legacy codepaths as deprecated? I think my previous constant initialization of period defaulting was not very good for our application. Thanks!

Reply:
I deleted the unused ones